src/internal/symbol/iterator.ts

```diff
@@ -1,38 +1,16 @@
-import { root } from '../util/root';
 
-export function symbolIteratorPonyfill(root: any) {
-  const Symbol: any = root.Symbol;
-
-  if (typeof Symbol === 'function') {
-    if (!Symbol.iterator) {
-      Symbol.iterator = Symbol('iterator polyfill');
-    }
-    return Symbol.iterator;
-  } else {
-    // [for Mozilla Gecko 27-35:](https://mzl.la/2ewE1zC)
-    const { Set } = root;
-    if (Set && typeof new Set()['@@iterator'] === 'function') {
-      return '@@iterator';
-    }
-    const { Map } = root;
-    // required for compatability with es6-shim
-    if (Map) {
-      let keys = Object.getOwnPropertyNames(Map.prototype);
-      for (let i = 0; i < keys.length; ++i) {
-        let key = keys[i];
-        // according to spec, Map.prototype[@@iterator] and Map.orototype.entries must be equal.
-        if (key !== 'entries' && key !== 'size' && Map.prototype[key] === Map.prototype['entries']) {
-          return key;
-        }
-      }
-    }
-    return '@@iterator';
-  }
+/* NOTE: Warning users that they don't have a Symbol.iterator
+   polyfill. We don't want to throw on this, because it's not required
+   by the library. However it will provide clues to users on older
+   browsers why things like `from(iterable)` doesn't work. */
+if (!Symbol || !Symbol.iterator) {
```

Comment:
@benlesh This still has the "Symbol is not defined" problem. Needed to use typeof checks. Here's a follow up PR to fix: #3394